### DISCUSSION TITLE
How to set yaw misalignment angle in OPENFAST input files

### DISCUSSION THREAD

**Discussion by sjihf12** [OpenFAST/openfast#1837] [Q&A]
In OPENFAST input files, which parameter represent the yaw misalignment angle？ I want to set yaw misalignment angle to 8 degrees to suit DLC1.6

**Comment by jjonkman** [comment]
Dear @sjihf12,

Yaw misalignment is the angle between the wind direction and nacelle-yaw angle.  So, you can set the yaw misalignment either by changing the wind direction (`PropagationDir` in the InflowWind module) or by changing the nacelle-yaw angle (`NacYaw` in the ElastoDyn module).  Note that if you change `NacYaw` and the nacelle-yaw degree of freedom of ElastoDyn is enabled (`YawDOF` = TRUE), you should also set `YawNeut` in the ServoDyn module equal to `NacYaw`.

Best regards,

**Reply by sjihf12** [reply]
Thank you!

**Comment by sjihf12** [comment]
Dear Professor
Does OPENFAST can output the blade root stress?  In flexure formula (sigma = M*y/Ixx),  the sigma is blade root stress and I know the OPENFAST can output M (blade root moment).

**Reply by jjonkman** [reply]
Dear @sjihf12,

OpenFAST will output the sectional loads (shear and axial forces, bending and torsion moments), but does not directly output stress or strain.  The latter would require post-processing via a sectional analysis tool.  Simple analytical formulae for stress from load (such as sigma = M*y/Ixx) can be applied for simple stress states of simple cross sections composed of isotropic materials, e.g., in the shaft and tower, but will likely not work well for complex blade cross sections composed of composite materials.

Best regards,

**Reply by sjihf12** [reply]
Thank you!

**Comment by sjihf12** [comment]
Dear Professor
I want to product the DLC1.6 condition "severe sea state (SSS) with 50 years recurrence period, and irregular waves of 10.4 m significant wave height, 14.7 s peak period, and seed number 14?". I know how to set significant wave height, peak period, and seed number in the HydroDyn. I do not know how to achieve “severe sea state (SSS) with 50 years recurrence period” in the HydroDyn.

**Reply by jjonkman** [reply]
Dear @sjihf12,

If you have calculated your site-specific values of Hs and Tp for the severe sea state (SSS), you can specify them in HydroDyn just like you would with any other irregular sea state.

Best regards,

**Reply by sjihf12** [reply]
Thank you

**Comment by Chris195513** [comment]
Dear @jjonkman 
I'm trying to simulate DLC 1.8 which considers an ice formation on all rotor blades and ice formation on all rotor blades except one (ice formation with a thickness of 30mm),
My question is; 
How to set ice conditions in openFAST input files?

**Reply by jjonkman** [reply]
Dear @Chris195513,

I'm not familiar with "DLC 1.8", but regarding the modeling of ice accretion on blades in OpenFAST, see a similar question asked and answered on our forum: https://forums.nrel.gov/t/atmospheric-ice-accretion-on-the-blades/908.

Best regards,

**Comment by Chris195513** [comment]
Dear @jjonkman 

Thank you for your reply, I will check that forum. I have another question, I want to simulate "Gust", Is it necessary to modify some parameters inside Inflow data? , Are some documents from NREL related to how to simulate the DLC?

**Reply by jjonkman** [reply]
Dear @Chris195513,

You can use the uniform wind files supported by the InflowWind via `WindType` = 2 to specify deterministic gust events.  You can use the old IECWind utility to generate these uniform wind data files for the deterministic gust events (ECG, EOG, etc.) defined within the IEC design standards.

Best regards,

**Comment by EWEMkite** [comment]
Dear @jjonkman,

I am trying to set wave and wind misalignment in my simulation according to IEC 61400-3 DLC 6.1

Refering to the post here: [https://forums.nrel.gov/t/is-positive-yaw-error-clockwise/582/18](url) which maintains wave propogation as 0 deg, can I rather maintain the wind direction as 0 deg for easier yaw error setting? So I only change the wave direction for wind wave misalignment.

Attached are my SeaState, Inflow and ElastoDyn files for DLC 6.1 with yaw error of 8°, wind-wave misalignment=30°. Could you please help me have a check? Thanks! 
[ElastoDyn.txt](https://github.com/user-attachments/files/19623731/ElastoDyn.txt)
[SeaState.txt](https://github.com/user-attachments/files/19623732/SeaState.txt)
[Inflow.txt](https://github.com/user-attachments/files/19623733/Inflow.txt)

Regards

**Reply by jjonkman** [reply]
Dear @EWEMkite,

Yes, you can certainly do what you propose, keeping the wind direction at zero, misaligning the waves by 30 degrees, and imposing a yaw error of 8 degrees.

Note that if you set `NacYaw` = 8 degrees with `YawDOF` = TRUE in ElastoDyn, you should set `YawNeut` in ServoDyn to 8 degrees as well; otherwise, the yaw spring in ServoDyn will quickly yaw the nacelle back to `YawNeut`.

Best regards,
